app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileScreen.kt

```diff
@@ -186,71 +189,77 @@ private fun CurrentSelfUserStatus(
+        ProfileStatusButton(
+            userStatus = UserStatus.AVAILABLE,
+            modifier = Modifier.weight(weight = 6F),
+            onClick = { status -> onStatusClicked(status) },
             text = stringResource(R.string.user_profile_status_available),
-            fillMaxWidth = false,
-            minHeight = dimensions().userProfileStatusBtnHeight,
-            state = if (userStatus == UserStatus.AVAILABLE) WireButtonState.Selected else WireButtonState.Default,
+            currentStatus = userStatus,
             shape = RoundedCornerShape(
                 topStart = dimensions().corner16x,
                 bottomStart = dimensions().corner16x
             ),
-            leadingIcon = {
-                UserStatusIndicator(
-                    status = UserStatus.AVAILABLE,
-                    modifier = Modifier.padding(end = dimensions().spacing4x).testTag("Available")
-                )
-            }
         )
-        WireSecondaryButton(
-            onClick = { onStatusClicked(UserStatus.BUSY) },
+        ProfileStatusButton(
+            userStatus = UserStatus.BUSY,
+            modifier = Modifier.weight(weight = 4F),
+            onClick = { status -> onStatusClicked(status) },
             text = stringResource(R.string.user_profile_status_busy),
-            fillMaxWidth = false,
-            minHeight = dimensions().userProfileStatusBtnHeight,
-            state = if (userStatus == UserStatus.BUSY) WireButtonState.Selected else WireButtonState.Default,
-            shape = RoundedCornerShape(0.dp),
-            leadingIcon = {
-                UserStatusIndicator(
-                    status = UserStatus.BUSY,
-                    modifier = Modifier.padding(end = dimensions().spacing4x).testTag("Busy")
-                )
-            }
+            currentStatus = userStatus,
         )
-        WireSecondaryButton(
-            onClick = { onStatusClicked(UserStatus.AWAY) },
+        ProfileStatusButton(
+            userStatus = UserStatus.AWAY,
+            modifier = Modifier.weight(weight = 4F),
+            onClick = { status -> onStatusClicked(status) },
             text = stringResource(R.string.user_profile_status_away),
-            fillMaxWidth = false,
-            minHeight = dimensions().userProfileStatusBtnHeight,
-            state = if (userStatus == UserStatus.AWAY) WireButtonState.Selected else WireButtonState.Default,
-            shape = RoundedCornerShape(0.dp),
-            leadingIcon = {
-                UserStatusIndicator(
-                    status = UserStatus.AWAY,
-                    modifier = Modifier.padding(end = dimensions().spacing4x).testTag("Away")
-                )
-            }
+            currentStatus = userStatus,
         )
-        WireSecondaryButton(
-            onClick = { onStatusClicked(UserStatus.NONE) },
+        ProfileStatusButton(
+            userStatus = UserStatus.NONE,
+            modifier = Modifier.weight(weight = 3F),
```

Comment:
I'm wondering if this fixed values `3F`, `4F`, `6F`, will resist all major cases 🤔  
Even tough, we currently not support landscape mode, will be a good time to do a little test before committing to these values. What do you think ?

Reply:
Unfortunately not all issues. Does in future we will have other languages? I Can make button weight dependable on text size

Reply:
Yes! In fact we currently support l10n 😅  we'll have other languages on those buttons